archspec/cpu/detect.py

```diff
@@ -88,24 +88,69 @@ def _machine():
     operating_system = platform.system()
 
     # If we are not on Darwin, trust what Python tells us
-    if operating_system != "Darwin":
+    if operating_system not in ("Darwin", "Windows"):
         return platform.machine()
 
-    # On Darwin it might happen that we are on M1, but using an interpreter
-    # built for x86_64. In that case "platform.machine() == 'x86_64'", so we
-    # need to fix that.
-    #
-    # See: https://bugs.python.org/issue42704
+    if operating_system == "Darwin":
+        # On Darwin it might happen that we are on M1, but using an interpreter
+        # built for x86_64. In that case "platform.machine() == 'x86_64'", so we
+        # need to fix that.
+        #
+        # See: https://bugs.python.org/issue42704
+        output = _check_output(
+            ["sysctl", "-n", "machdep.cpu.brand_string"], env=_ensure_bin_usrbin_in_path()
```

Comment:
@markus-ferrell the CI appears to be complaining about this line length: can you reduce it to <80 characters?